spring-integration-jdbc/src/main/java/org/springframework/integration/jdbc/JdbcPollingChannelAdapter.java

```diff
@@ -200,8 +205,32 @@ protected Object doReceive() {
 	}
 
 	private void executeUpdateQuery(Object obj) {
-		SqlParameterSource updateParameterSource = this.sqlParameterSourceFactory.createParameterSource(obj);
-		this.jdbcOperations.update(this.updateSql, updateParameterSource);
+		this.jdbcOperations.update(this.updateSql, this.sqlParameterSourceFactory.createParameterSource(obj));
+	}
+
+	private static final class PreparedStatementCreatorWithMaxRows implements PreparedStatementCreator, SqlProvider {
```

Comment:
Shouldn't we implement all the delgate's interfaces?

```java
	private class PreparedStatementCreatorImpl
			implements PreparedStatementCreator, PreparedStatementSetter, SqlProvider, ParameterDisposer {
```

I see code like this in the template...

```java
			finally {
				if (pss instanceof ParameterDisposer) {
					((ParameterDisposer) pss).cleanupParameters();
				}
			}
```